### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by kramstrom**

### CODE REVIEW

File: frontend/src/helpers/velocityAnimate.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import Velocity from 'velocity-animate'
+
+export const lineBeforeEnter = (el, x1, y1, x2, y2) => {
```
Comment (antoncarlsson) [inline frontend/src/helpers/velocityAnimate.js]: Seemingly duplicate code in this file, could be refactored?

File: frontend/src/store/modules/news.js [hunk 2]
```diff
@@ -96,6 +103,7 @@ const actions = {
       datetime: convertDateStringToDateObj(news.datetime)
     }
 
+    let locationIds = {}
     let location = { ...news.location }
     for (const key of Object.keys(location)) {
```
Comment (antoncarlsson) [inline frontend/src/store/modules/news.js]: Use forEach instead.

File: frontend/src/components/Notifications/NotificationsCity.vue [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    animations.circleEnter(circle, this.city.x, this.city.y)
+    animations.textBeforeEnter(text, this.city.municipalityX, this.city.municipalityY)
+    animations.textEnter(text, this.city.x, this.city.y)
+  },
+  beforeDestroyed: function () {
+    const line = this.$refs['cityLine-' + this.city.name]
+    const circle = this.$refs['cityCircle-' + this.city.name]
+    const text = this.$refs['cityText-' + this.city.name]
+
+    animations.lineLeave(line, this.city.municipalityX, this.city.municipalityY, this.city.municipalityX, this.city.municipalityY)
```
Comment (antoncarlsson) [inline frontend/src/components/Notifications/NotificationsCity.vue]: Line too long.

File: frontend/src/components/Notifications/NotificationsCity.vue [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+export default {
+  name: 'NotificationsCity',
+  props: ['circleSize', 'fontSize', 'yOffset', 'lineWidth', 'city'],
+  mounted: function () {
+    const line = this.$refs['cityLine-' + this.city.name]
+    const circle = this.$refs['cityCircle-' + this.city.name]
+    const text = this.$refs['cityText-' + this.city.name]
+
+    animations.lineBeforeEnter(line, this.city.municipalityX, this.city.municipalityY, this.city.municipalityX, this.city.municipalityY)
```
Comment (antoncarlsson) [inline frontend/src/components/Notifications/NotificationsCity.vue]: Line too long.

File: frontend/src/store/modules/news.js (hunk 2 above)
Comment (kramstrom) [reply frontend/src/store/modules/news.js]: Yes sir

File: frontend/src/components/Notifications/NotificationsCity.vue (hunk 4 above)
Comment (kramstrom) [reply frontend/src/components/Notifications/NotificationsCity.vue]: Yes sir

File: frontend/src/components/Notifications/NotificationsCity.vue (hunk 3 above)
Comment (kramstrom) [reply frontend/src/components/Notifications/NotificationsCity.vue]: Ye boy

File: frontend/src/components/Notifications/NotificationsMunicipality.vue [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    animations.circleEnter(circle, this.municipality.x, this.municipality.y)
+    animations.textBeforeEnter(text, this.municipality.countyX, this.municipality.countyY)
+    animations.textEnter(text, this.municipality.x, this.municipality.y)
+  },
+  beforeDestroyed: function () {
+    const line = this.$refs['municipalityLine-' + this.municipality.name]
+    const circle = this.$refs['municipalityCircle-' + this.municipality.name]
+    const text = this.$refs['municipalityText-' + this.municipality.name]
+
+    animations.lineLeave(line, this.municipality.countyX, this.municipality.countyY, this.municipality.countyX, this.municipality.countyY)
```
[the remark this answers is not in the document]
Comment (kramstrom) [reply frontend/src/components/Notifications/NotificationsMunicipality.vue]: Ayay

File: frontend/src/components/Notifications/NotificationsMunicipality.vue [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+export default {
+  name: 'NotificationsMunicipality',
+  props: ['circleSize', 'fontSize', 'yOffset', 'lineWidth', 'municipality'],
+  mounted: function () {
+    const line = this.$refs['municipalityLine-' + this.municipality.name]
+    const circle = this.$refs['municipalityCircle-' + this.municipality.name]
+    const text = this.$refs['municipalityText-' + this.municipality.name]
+
+    animations.lineBeforeEnter(line, this.municipality.countyX, this.municipality.countyY, this.municipality.countyX, this.municipality.countyY)
```
[the remark this answers is not in the document]
Comment (kramstrom) [reply frontend/src/components/Notifications/NotificationsMunicipality.vue]: Yee

File: frontend/src/helpers/velocityAnimate.js (hunk 1 above)
Comment (kramstrom) [reply frontend/src/helpers/velocityAnimate.js]: We considered this but decided that it would reduce readability of code

**Pull request merged**